### PR TITLE
Support oauth token endpoint, add wiki verification support, fix issue #420

### LINKED ISSUE #420: Heading blocks do not have the `is_toggleable` property in their types
[The specification says](https://developers.notion.com/reference/block#headings) that `heading_1`, `heading_2`, and `heading_3` blocks should have an `is_toggleable` property within the `heading_x` property. I have verified against the API that it is reporting this data correctly (and therefore returning it through the client), but it is not typed correctly in the library.

While it is present inside [`BlockObjectRequest`](https://github.com/makenotion/notion-sdk-js/blob/550c51c68ed7b6c077ab376c5535af0faed2d54f/src/api-endpoints.ts#L6690), [`BlockObjectRequestWithoutChildren`](https://github.com/makenotion/notion-sdk-js/blob/550c51c68ed7b6c077ab376c5535af0faed2d54f/src/api-endpoints.ts#L6067), [`UpdateBlockBodyParameters`](https://github.com/makenotion/notion-sdk-js/blob/550c51c68ed7b6c077ab376c5535af0faed2d54f/src/api-endpoints.ts#L10020), it is not present inside `BlockObjectResponse` (specifically [`Heading1BlockObjectResponse`](https://github.com/makenotion/notion-sdk-js/blob/550c51c68ed7b6c077ab376c5535af0faed2d54f/src/api-endpoints.ts#L4792), `Heading2BlockObjectResponse`, and `Heading3BlockObjectResponse`). I’d normally submit a PR but I’m aware that `api-endpoints.ts` is auto-generated; I’ve patched the library for my own usage.

This is somewhat related to #245, although I believe @aalemayhu was using a version of the API before the `is_toggleable` property was included in the response.

### PR DESCRIPTION
**Pull request by rhart92**
Fixes #420 and #73.

### CODE REVIEW

File: src/api-endpoints.ts [hunk 1]
```diff
@@ -688,6 +688,57 @@ type FormulaPropertyResponse =
   | NumberFormulaPropertyResponse
   | BooleanFormulaPropertyResponse
 
+type VerificationPropertyUnverifiedResponse = {
+  state: "unverified"
+  date: null
+  verified_by: null
+}
+
+type VerificationPropertyResponse = {
```
[the remark this answers is not in the document]
Comment (rhart92) [reply src/api-endpoints.ts:697]: Yes

File: src/Client.ts [hunk 2]
```diff
@@ -525,6 +553,29 @@ export default class Client {
     })
   }
 
+  public readonly oauth = {
```
[the remark this answers is not in the document]
Comment (rhart92) [reply src/Client.ts:556]: Allows you to call the `token` endpoint.

```ts
const { Client } = require("@notionhq/client")

// Initializing a client
const notion = new Client({ ... })

const tokenResponse = await notion.oauth.token({ ... })
```

This is similar to how we nest all the `block` methods under a `blocks` object.

[the remark this answers is not in the document]
Comment (hazel-notion) [reply src/Client.ts:556]: hmm... we were not allowed to call the `token` endpoint before your change?

[the remark this answers is not in the document]
Comment (rhart92) [reply src/Client.ts:556]: Not through the SDK 😓 See https://github.com/makenotion/notion-sdk-js/issues/73

**Review by hazel-notion — APPROVED** [review]

**Pull request merged**